### PR TITLE
Feature/exception catcher

### LINKED ISSUE #56: Prompt and Capture Exception
Install this package: https://www.npmjs.com/package/react-native-exception-handler
When some exception happened and whole app crashed, will be smoother for users to do bug reporting

### PR DESCRIPTION
**Pull request by tsejerome**
Resolves #56

### CODE REVIEW

File: utils/sendMail.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  bcc?: Email;
+};
+
+export async function sendErrorCrashEmail(
+  e: Error | string,
+  isFatal: boolean = true,
+) {
+  let error = typeof e === 'string' ? e : `${e.name} ${e.message}`;
+  sendEmail(
+    'tsejerome1997@gmail.com',
```
Comment (aguycalled) [inline utils/sendMail.ts:17]: shouldn't this be updated to dev@whisperwallet.net?

File: utils/sendMail.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    'My Whisper Wallet crashed!',
+    `Hello Whisper Development Team, <br>My Whisper Wallet got some issues. <br>Here is my The error is as below: <br>Error:
```
Comment (aguycalled) [inline utils/sendMail.ts:19]: fix typos here

File: App.tsx [hunk 3]
```diff
@@ -45,14 +49,100 @@ win.indexedDB.__useShim();
...
+    'crashErrorRecords',
+    '',
+  );
+
   const [shownWelcome, setShownWelcome] = useState('false');
 
+  const previousNativeErrorHandler = (errorMessage: string) => {
+    Alert.alert(
+      'There was an unexpected error occurred in the previous app launch',
+      'Do you want to send a report to Whisper Team?',
```
Comment (aguycalled) [inline App.tsx:113]: the Whisper development team

**Comment by aguycalled** [comment]
crashes on both iOS device and simulator when trying to send email or copying to clipboard

File: App.tsx (hunk 3 above)
Comment (aguycalled) [reply App.tsx:113]: There was an unexpected error during the previous session.
Do you want to send a report to the Whisper development team?

**Comment by aguycalled** [comment]
if an exception happens while the blurry modal is shown, then the toast is not visible:

image

**Comment by aguycalled** [comment]
possible solution: https://github.com/calintamas/react-native-toast-message/blob/main/docs/modal-usage.md

**Comment by aguycalled** [comment]
> if an exception happens while the blurry modal is shown, then the toast is not visible:
> 
> image

fixed in https://github.com/navcoin/WhisperWallet/pull/104/commits/1db374948ab10f9f573c91b3a90fb4fea6e9c047

**Comment by aguycalled** [comment]
image

the toast text is not readable
tapping the toast does nothing
the toast does not stay visible long enough

what happens if I miss the toast? it should be possible to access the exceptions log from somewhere (maybe settings?)

**Comment by aguycalled** [comment]
more comments:

- the toast does not seem to have the background colour we commented on telegram (aligned with the app ui)
- I think, cosmetically, it would benefit from a bit more of space on top of the toast

**Comment by aguycalled** [comment]
> more comments:
> 
> * the toast does not seem to have the background colour we commented on telegram (aligned with the app ui)
> * I think, cosmetically, it would benefit from a bit more of space on top of the toast

those and these:

> also, the text says something was wrong in the previous session, but this exception just occurred now.

are fixed with https://github.com/navcoin/WhisperWallet/pull/104/commits/13774a7a062e35a054944f713ac8a0470a1c5d1f and https://github.com/navcoin/WhisperWallet/pull/104/commits/5a1f9c36d5fada698c963b1eaf580a1d83631902

**Comment by tsejerome** [comment]
https://user-images.githubusercontent.com/20391660/165784136-68c9c3d7-a3fb-4bef-988c-3090de044d64.mp4

Ready for review again

**Comment by aguycalled** [comment]
everytime I restart the app on this branch I am shown the onboarding screen and asked if I want to lock the screen. seems like something related to AsyncStorage

**Comment by tsejerome** [comment]
Just resolved the issue you mentioned! its about RNBootsplash.hide and the setLoaded in App.tsx

**Comment by tsejerome** [comment]
@aguycalled I just resolved some merge conflict for this branch, please review and see if there is any issues (if not please kindly merge)

**Comment by aguycalled** [comment]
merge looks good, tested on simulator

**Review by aguycalled — APPROVED** [review]

**Pull request merged**